core/src/main/java/org/elasticsearch/discovery/local/LocalDiscovery.java

```diff
@@ -305,6 +285,11 @@ public DiscoveryStats stats() {
     }
 
     @Override
+    public DiscoverySettings getDiscoverySettings() {
+        return discoverySettings;
+    }
+
```

Comment:
Yeah, I was just wondering why we need a dedicated `DiscoverySettings` class. In other places we just pass `ClusterSettings` around.